src/getChildEventSubscriber.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * @flow
+ */
+
+import type {
+  NavigationEventSubscriber,
+  NavigationAction,
+  NavigationState,
+  NavigationEventPayload,
+} from './TypeDefinition';
+
+/*
+ * This is used to extract one children's worth of events from a stream of navigation action events
+ *
+ * Based on the 'action' events that get fired for this navigation state, this utility will fire
+ * focus and blur events for this child
+ */
+export default function getChildEventSubscriber(
+  addListener: NavigationEventSubscriber,
+  key: string
+): NavigationEventSubscriber {
+  const actionSubscribers = new Set();
+  const willFocusSubscribers = new Set();
+  const didFocusSubscribers = new Set();
+  const willBlurSubscribers = new Set();
+  const didBlurSubscribers = new Set();
+
+  const getChildSubscribers = evtName => {
+    switch (evtName) {
+      case 'action':
+        return actionSubscribers;
+      case 'willFocus':
+        return willFocusSubscribers;
+      case 'didFocus':
+        return didFocusSubscribers;
+      case 'willBlur':
+        return willBlurSubscribers;
+      case 'didBlur':
+        return didBlurSubscribers;
+      default:
+        return null;
+    }
+  };
+
+  const emit = (eventName, payload) => {
+    const subscribers = getChildSubscribers(eventName);
+    subscribers &&
+      subscribers.forEach(subs => {
```

Comment:
nit: `subs` sounds plural and it tripped me up a bit, might be better to make it `sub` or `subscriber`